iosApp/iosApp/Explore/RepoListView.swift

```diff
@@ -7,23 +7,68 @@
 //
 
 import SwiftUI
+import ExploreShared
 
 struct RepoListView: View {
     @StateObject var viewModel = RepoListViewModel()
 
     var body: some View {
-        ForEach(self.viewModel.repoList, id: \.name) { repo in
-            VStack {
-                Text(repo.name)
-            }.padding(.bottom, 10)
-        }.onAppear {
-            Task {
-                await self.viewModel.loadRepos()
+        let list = self.viewModel.repoList
+        let isLastPage = self.viewModel.isLastPage
+
+        ZStack {
+            ScrollView(.vertical) {
+                LazyVStack {
+                    if !list.isEmpty {
+                        ListView(list)
+                    }
+                    
+                    if !isLastPage {
+                        LoadingView()
+                            .onAppear {
+                                Task {
+                                    await self.viewModel.loadRepos()
+                                }
+                            }
+                    }
+                }
             }
         }
     }
 }
 
+private struct ListView: View {
+    private var list: [Repo]
+    
+    init(_ list: [Repo]) {
+        self.list = list
+    }
+    
+    var body: some View {
+        ForEach(list, id: \.name) { repo in
+            RepoView(repo)
+        }
+    }
+}
+
+private struct LoadingView: View {
```

Comment:
I think we can extract this into a separate file and reuse it in other future views. We can have a `Common > Components` directory. What do you think?